src/js/components/SelectMultiple/SelectMultiple.js

```diff
@@ -152,6 +152,23 @@ const SelectMultiple = forwardRef(
       if (!search) setAllOptions(optionsProp);
     }, [optionsProp, search]);
 
+    useEffect(() => {
+      if (search && optionsProp && optionsProp.length > 0) {
+        const additionalOptions = [...allOptions];
+        optionsProp.forEach((i) =>
+          additionalOptions.some((j) => {
+            if (typeof i === 'object')
```

Comment:
Can this be rewritten to:

```
return typeof i === 'object' ? applyKey(i, valueKey) === applyKey(j, valueKey) : i === j;
```

Reply:
Although if this feels less readable, i'm fine leaving the initial implementation

Reply:
Incorporated this in the latest commit. I think you suggestion doesn't hurt readability